src/AnimatedLineGraph.tsx

```diff
@@ -193,17 +199,25 @@ export function AnimatedLineGraph({
         damping: 50,
         velocity: 0,
       })
-      if (!active) pathEnd.current = 1
+      if (!active && resetPositionOnRelease) pathEnd.current = 1
 
       if (active) onGestureStart?.()
       else onGestureEnd?.()
     },
-    [circleRadius, onGestureEnd, onGestureStart, pathEnd]
+    [
+      circleRadius,
+      onGestureEnd,
+      onGestureStart,
+      pathEnd,
+      resetPositionOnRelease,
+    ]
   )
   useAnimatedReaction(
-    () => x.value,
-    (fingerX) => {
-      runOnJS(setFingerX)(fingerX)
+    () => [x.value, isActive.value],
+    ([fingerX, isActiveValue]) => {
+      if (isActiveValue) {
+        runOnJS(setFingerX)(fingerX as number)
+      }
```

Comment:
Why this change? Afaik it worked fine before

Reply:
This goes with the idea of the graph highlight & the initialIndex. On the initial render, the setFingerX function is executed, but as there is no gesture, the highlight path starts at 0.